geoutils/geovector.py

```diff
@@ -162,15 +165,31 @@ def create_mask(
             transform = rio.transform.from_bounds(left, bottom, right, top, width, height)
 
         # otherwise use directly rst's dimensions
+        elif isinstance(rst, gu.Raster):
+            out_shape = rst.shape
+            transform = rst.transform
+            crs = rst.crs
+            bounds = rst.bounds
         else:
-            out_shape = rst.shape  # type: ignore
-            transform = rst.transform  # type: ignore
-            crs = rst.crs  # type: ignore
+            raise ValueError("`rst` must be either a str, geoutils.Raster or None")
+
+        # Crop vector geometries to avoid issues when reprojecting
+        left, bottom, right, top = bounds  # type: ignore
+        x1, y1, x2, y2 = warp.transform_bounds(crs, self.ds.crs, left, bottom, right, top)
+        self.ds = self.ds.cx[x1:x2, y1:y2]
 
         # Reproject vector into rst CRS
         # Note: would need to check if CRS are different
```

Comment:
I know this wasn't added now, but GeoPandas checks the CRS so there's no need for us to do that as well. This comment can be removed.

Reply:
What do you mean? I added this step because I had issues after reprojecting certain RGI outlines to a local UTM zone. Some geometries turned into inf values.

Reply:
I mean the comment! If a `GeoDataFrame.to_crs()` is called with the same CRS as it currently has, it will just skip any computations. Therefore, there's no need to check if the CRSs are different, because the call below will reproject if needed, and won't if not needed.

Reply:
Oh ok, I thought you meant the lines above... Thanks for the clarification, will delete then!